### PR TITLE
cleanup code after rust 1.61 release

### PR DESCRIPTION
**Pull request by bluejekyll**

### CODE REVIEW

**Review by djc — COMMENTED** [review]
The changes here look okay to me, I'm a little confused about the context.

File: crates/server/src/authority/zone_type.rs [hunk 1]
```diff
@@ -5,7 +5,7 @@
 // http://opensource.org/licenses/MIT>, at your option. This file may not be
 // copied, modified, or distributed except according to those terms.
 
-#![allow(deprecated)]
+#![allow(deprecated, clippy::use_self)]
```
[the remark this answers is not in the document]
Comment (bluejekyll) [reply crates/server/src/authority/zone_type.rs:8]: > So why allow use_self?

It looks like in 1.61 some this clippy lint became more "accurate" in that it's warning on Self not being used in the context of derived interfaces on structs.

> And why is this PR predicated on the 1.61 release?

The use_self lint became a bit more accurate it appears, catching a bunch of cases it didn't before.

Honestly, I could just as easily be convinced that this is not a lint that we want to use given how pedantic it is (in the case of Derived interfaces).

[the remark this answers is not in the document]
Comment (djc) [reply crates/server/src/authority/zone_type.rs:8]: Can you give me an example of where you think it's wrong?

(I think I already mentioned I'm a fan of the lint, so would be good to apply it where possible IMO.)

[the remark this answers is not in the document]
Comment (bluejekyll) [reply crates/server/src/authority/zone_type.rs:8]: Using `ZoneType` as an example, if you remove the allow `clippy::use_self`:

```text
error: unnecessary structure name repetition
  --> crates/server/src/authority/zone_type.rs:14:10
   |
14 | pub enum ZoneType {
   |          ^^^^^^^^ help: use the applicable keyword: `Self`
   |
   = note: `-D clippy::use-self` implied by `-D warnings`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#use_self
```

I believe this is something that's being generated in the derive for `Serialize`, and/or `Deserialize`. I recently patched my `enum-as-inner` crate for that reason (it was generating code referencing the enum type name in the generated functions). I haven't had a chance to dive into Serde though.

[the remark this answers is not in the document]
Comment (bluejekyll) [reply crates/server/src/authority/zone_type.rs:8]: @djc, any concerns with merging this?

[the remark this answers is not in the document]
Comment (djc) [reply crates/server/src/authority/zone_type.rs:8]: Nope!

**Review by djc — APPROVED** [review]

**Pull request merged**